File: library/src/main/java/nl/endevelopment/r2randroid/r2rlib/models/IndicativePrice.java
package nl.endevelopment.r2randroid.r2rlib.models;

import android.os.Parcel;
import android.os.Parcelable;

/**
 *If the estimated price is a single value (not a range) then only price is returned.
 If the estimated price is a range, then both price and priceLow/priceHigh are returned.
 You can choose to display either the single price (for brevity) or the range (for accuracy).
 */
public class IndicativePrice implements Parcelable {

    /**
     name 	string 	Display name of price level (optional)
     price 	float 	Median price
     priceLow 	float 	Lower bound of price range (optional)
     priceHigh 	float 	Upper bound of price range (optional)
     currency 	string 	Currency code (ISO 4217)
     nativePrice 	float 	Median price in native currency (optional)
     nativePriceLow 	float 	Lower bound of price range in native currency (optional)
     nativePriceHigh 	float 	Upper bound of price range in native currency (optional)
     nativeCurrency 	string 	Native currency code (ISO 4217) (optional)
     */

    private String name;
    private float price;
    private float priceLow;
    private float priceHigh;
    private String currency;
    private float nativePrice;
    private float nativePriceLow;
    private float nativePriceHigh;
    private String nativeCurrency;

    public IndicativePrice() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public float getPrice() {
        return price;
    }

    public void setPrice(float price) {
        this.price = price;
    }

    public float getPriceLow() {
        return priceLow;
    }

    public void setPriceLow(float priceLow) {
        this.priceLow = priceLow;
    }

    public float getPriceHigh() {
        return priceHigh;
    }

    public void setPriceHigh(float priceHigh) {
        this.priceHigh = priceHigh;
    }

    public String getCurrency() {
        return currency;
    }

    public void setCurrency(String currency) {
        this.currency = currency;
    }

    public float getNativePrice() {
        return nativePrice;
    }

    public void setNativePrice(float nativePrice) {
        this.nativePrice = nativePrice;
    }

    public float getNativePriceLow() {
        return nativePriceLow;
    }

    public void setNativePriceLow(float nativePriceLow) {
        this.nativePriceLow = nativePriceLow;
    }

    public float getNativePriceHigh() {
        return nativePriceHigh;
    }

    public void setNativePriceHigh(float nativePriceHigh) {
        this.nativePriceHigh = nativePriceHigh;
    }

    public String getNativeCurrency() {
        return nativeCurrency;
    }

    public void setNativeCurrency(String nativeCurrency) {
        this.nativeCurrency = nativeCurrency;
    }

    protected IndicativePrice(Parcel in) {
        name = in.readString();
        price = in.readFloat();
        priceLow = in.readFloat();
        priceHigh = in.readFloat();
        currency = in.readString();
        nativePrice = in.readFloat();
        nativePriceLow = in.readFloat();
        nativePriceHigh = in.readFloat();
        nativeCurrency = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeFloat(price);
        dest.writeFloat(priceLow);
        dest.writeFloat(priceHigh);
        dest.writeString(currency);
        dest.writeFloat(nativePrice);
        dest.writeFloat(nativePriceLow);
        dest.writeFloat(nativePriceHigh);
        dest.writeString(nativeCurrency);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<IndicativePrice> CREATOR = new Parcelable.Creator<IndicativePrice>() {
        @Override
        public IndicativePrice createFromParcel(Parcel in) {
            return new IndicativePrice(in);
        }

        @Override
        public IndicativePrice[] newArray(int size) {
            return new IndicativePrice[size];
        }
    };

    @Override
    public String toString() {
        return "IndicativePrice{" +
                "name='" + name + '\'' +
                ", price=" + price +
                ", priceLow=" + priceLow +
                ", priceHigh=" + priceHigh +
                ", currency='" + currency + '\'' +
                ", nativePrice=" + nativePrice +
                ", nativePriceLow=" + nativePriceLow +
                ", nativePriceHigh=" + nativePriceHigh +
                ", nativeCurrency='" + nativeCurrency + '\'' +
                '}';
    }
}

File: library/src/main/java/com/getroadmap/r2rlib/models/IndicativePrice.kt
package com.getroadmap.r2rlib.models

import android.os.Parcel
import android.os.Parcelable

/**
 * If the estimated price is a single value (not a range) then only price is returned.
 * If the estimated price is a range, then both price and priceLow/priceHigh are returned.
 * You can choose to display either the single price (for brevity) or the range (for accuracy).
 */
open class IndicativePrice() : Parcelable {

    /**
     * name 	string 	Display name of price level (optional)
     * price 	float 	Median price
     * priceLow 	float 	Lower bound of price range (optional)
     * priceHigh 	float 	Upper bound of price range (optional)
     * currency 	string 	Currency code (ISO 4217)
     * nativePrice 	float 	Median price in native currency (optional)
     * nativePriceLow 	float 	Lower bound of price range in native currency (optional)
     * nativePriceHigh 	float 	Upper bound of price range in native currency (optional)
     * nativeCurrency 	string 	Native currency code (ISO 4217) (optional)
     */

    var name: String? = null
    var price: Float? = null
    var priceLow: Float? = null
    var priceHigh: Float? = null
    var currency: String? = null
    var nativePrice: Float? = null
    var nativePriceLow: Float? = null
    var nativePriceHigh: Float? = null
    var nativeCurrency: String? = null

    constructor(parcel: Parcel) : this() {
        name = parcel.readString()
        price = parcel.readValue(Float::class.java.classLoader) as? Float
        priceLow = parcel.readValue(Float::class.java.classLoader) as? Float
        priceHigh = parcel.readValue(Float::class.java.classLoader) as? Float
        currency = parcel.readString()
        nativePrice = parcel.readValue(Float::class.java.classLoader) as? Float
        nativePriceLow = parcel.readValue(Float::class.java.classLoader) as? Float
        nativePriceHigh = parcel.readValue(Float::class.java.classLoader) as? Float
        nativeCurrency = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeValue(price)
        parcel.writeValue(priceLow)
        parcel.writeValue(priceHigh)
        parcel.writeString(currency)
        parcel.writeValue(nativePrice)
        parcel.writeValue(nativePriceLow)
        parcel.writeValue(nativePriceHigh)
        parcel.writeString(nativeCurrency)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<IndicativePrice> {
        override fun createFromParcel(parcel: Parcel): IndicativePrice {
            return IndicativePrice(parcel)
        }

        override fun newArray(size: Int): Array<IndicativePrice?> {
            return arrayOfNulls(size)
        }
    }

}